client/src/ui/explore/OpportunityCard.tsx

```diff
@@ -1,9 +1,263 @@
+  Chip,
+  Button,
+  Avatar,
+  IconButton,
+} from "@mui/material";
+import {
+  faAnglesUp,
+  faAnglesDown,
+  faComment,
+  faReply,
+  faBookmark,
+} from "@fortawesome/free-solid-svg-icons";
+import { faBookmark as regularBookmark } from "@fortawesome/free-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { OpportunityTimeline } from "./OpportunityTimeline";
+import { useState } from "react";
 
+interface OpportunityCardProps {
+  id: number;
+  title: string;
+  jobPosition: string;
+  jobType: string;
+  jobAvatar: string;
+  positionStatus: boolean;
+  userPositionStatus: boolean;
+  totalApplied: number;
+  rejected: number;
+  oa: number;
+  interviewing: number;
+  offered: number;
+  recentMessages: number;
+  bookmarked: boolean;
+}
 
-export function OpportunityCard () {
-    return(
-        <div>
-            
+interface jobStats {
+  status: string;
+  color: string;
+  quantity: number;
+}
+
+export function OpportunityCard({
+  title,
+  jobPosition,
+  jobType,
+  jobAvatar,
+  positionStatus,
+  userPositionStatus,
+  totalApplied,
+  rejected,
+  oa,
+  interviewing,
+  offered,
+  recentMessages,
+  bookmarked: initialBookmarked,
+}: OpportunityCardProps) {
+  // To be replaced with actuall book mark update from DB
+  const [bookmarked, setBookmarked] = useState(initialBookmarked);
+
+  // Array for mapping the job stats
+  const stats: jobStats[] = [
+    {
+      status: "Total Applied:",
+      color: "black",
+      quantity: totalApplied,
+    },
+    {
+      status: "Rejected:",
+      color: "red",
+      quantity: rejected,
+    },
+    {
+      status: "OA:",
+      color: "orange",
+      quantity: oa,
+    },
+    {
+      status: "Interviewing:",
+      color: "gold",
+      quantity: interviewing,
+    },
+    {
+      status: "Offered:",
+      color: "green",
+      quantity: offered,
+    },
+  ];
+
+  const handleBookmark = () => {
+    setBookmarked((prev) => !prev); // Toggle the bookmark state
+  };
+
+  return (
+    <Card
+      style={{
+        border: "solid 3px #e0e4f2",
+        margin: "1.5rem 0",
+        borderRadius: "20px",
+        padding: ".5rem 2rem",
+      }}
+    >
+      {/* Card Header in a div with Bookmark */}
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "space-between",
+          alignItems: "center",
+          margin: 0,
+          padding: "1rem 0rem",
+        }}
+      >
+        <CardHeader
+          style={{ margin: 0, padding: ".5rem", height: "2rem" }}
+          avatar={<Avatar src={jobAvatar} style={{ margin: 0 }}></Avatar>}
+          title={<Typography variant="h5">{title}</Typography>}
+          subheader={
+            // Job Postion and Job Type in header
+            <Typography variant="h6">
+              {jobPosition}, {jobType}
+            </Typography>
+          }
+        />
+
+        {/* Bookmark button */}
+        <IconButton onClick={handleBookmark}>
```

Comment:
The background when hovering on the bookmark looks ugly.

Reply:
fixed